client/js/libs/handlebars/parse.js

```diff
@@ -1,125 +1,113 @@
-import Handlebars from "handlebars/runtime";
-import URI from "urijs";
-
-module.exports = function(text) {
-	text = Handlebars.Utils.escapeExpression(text);
-	text = colors(text);
-	text = channels(text);
-	text = uri(text);
-	return text;
-};
+const Handlebars = require("handlebars/runtime");
+const URI = require("urijs");
+const parseStyle = require("ircmessageparser/lib/parseStyle");
+const findChannels = require("ircmessageparser/lib/findChannels");
+const merge = require("ircmessageparser/lib/merge");
 
-function uri(text) {
-	return URI.withinString(text, function(url) {
-		if (url.indexOf("javascript:") === 0) {
-			return url;
-		}
-		var split = url.split("<");
-		url = "<a href='" + split[0].replace(/^www/, "http://www") + "' target='_blank' rel='noopener'>" + split[0] + "</a>";
-		if (split.length > 1) {
-			url += "<" + split.slice(1).join("<");
-		}
-		return url;
-	});
-}
 
-/**
- * Channels names are strings of length up to fifty (50) characters.
- * The only restriction on a channel name is that it SHALL NOT contain
- * any spaces (' '), a control G (^G or ASCII 7), a comma (',').
- * Channel prefix '&' is handled as '&amp;' because this parser is executed
- * after entities in the message have been escaped. This prevents a couple of bugs.
- */
-function channels(text) {
-	return text.replace(
-		/(^|\s|\x07|,)((?:#|&amp;)[^\x07\s,]{1,49})/g,
-		'$1<span class="inline-channel" role="button" tabindex="0" data-chan="$2">$2</span>'
-	);
-}
+const commonSchemes = [
+	"http", "https",
+	"ftp", "sftp",
+	"smb", "file",
+	"irc", "ircs",
+	"svn", "git",
+	"steam", "mumble", "ts3server", "ssh",
+];
```

Comment:
Wouldn't it quickly become an issue to whitelist schemes here? I feel like this is a well-known problem that should be handled in one of the libraries we use instead of in our code. Thoughts?